spyder/widgets/collectionseditor.py

```diff
@@ -1312,7 +1315,14 @@ def __init__(self, parent, data, readonly=False, title="", remote=False):
         else:
             self.editor = CollectionsEditorTableView(self, data, readonly,
                                                      title)
+        toolbar = SpyderToolbar(parent=None, title='Editor toolbar')
+
+        for item in self.editor.menu_actions:
+            if item is not None:
+                toolbar.addAction(item)
+

```

Comment:
Please add here the following line
```suggestion

# Update the toolbar actions state
self.editor.refresh_menu()

```

Also, please add to the `BaseTable.setup_table` method the following line:

```python
self.selectionModel().selectionChanged.connect(self.refresh_menu)
```

With this two lines I think we can achieve the actions update depending in what has been selected in the editor